ports/templates/ports/port-detail/build_information.html

```diff
@@ -0,0 +1,94 @@
+        <select onchange="filterBuilds()" class="form-control ml-2 filter" id="builder-filter" name="builder-filter">
+            <option value="">All</option>
+            {% for builder_option in builders %}
+                <option value="{{ builder_option }}" {%  if builder == builder_option %}selected{% endif %}>{{ builder_option }}</option>
+            {% endfor %}
+        </select>
+    </div>
+</div>
+<br><br>
+{% if builds|length > 0 %}
+<h4 class="text-secondary">Showing {% if status == "" %} <ins>All Builds</ins> {% else %}builds with status: <ins>{{ status }}</ins>{% endif %} on <ins>{% if builder == "" %}All Builders {% else %}{{ builder }}{% endif %}</ins></h4>
+
+{% if builds.has_other_pages %}
+            Page {{ builds.number }} of {{ builds.paginator.num_pages }} | Showing builds {{ builds.start_index }} to
+            {{ builds.end_index }}<br><br>
+            <nav aria-label="Page navigation">
+                <ul class="pagination">
+                    {% if builds.has_previous %}
+                        <li class="page-item"><button class="page-link" onclick="changePage({{ builds.previous_page_number }})">&laquo;</button></li>
+                    {% else %}
+                        <li class="disabled page-item"><span class="page-link">&laquo;</span></li>
+                    {% endif %}
+
+                    {% if builds.number|add:'-4' > 1 %}
+                        <li class="page-item"><button class="page-link" onclick=changePage({{ builds.number|add:'-5' }})>&hellip;</button>
+                        </li>
+                    {% endif %}
+
+                    {% for i in builds.paginator.page_range %}
+                        {% if builds.number == i %}
+                            <li class="active page-item"><span class="page-link">{{ i }} <span
+                                    class="sr-only">(current)</span></span></li>
+                        {% elif i > builds.number|add:'-5' and i < builds.number|add:'5' %}
+                            <li class="page-item"><button class="page-link" onclick="changePage({{ i }})">{{ i }}</button></li>
+                        {% endif %}
+                    {% endfor %}
+
+                    {% if builds.paginator.num_pages > builds.number|add:'4' %}
+                        <li class="page-item"><button class="page-link" onclick="changePage({{ builds.number|add:'5' }})">&hellip;</button>
+                        </li>
+                    {% endif %}
+
+                    {% if builds.has_next %}
+                        <li class="page-item"><button class="page-link" onclick="changePage({{ builds.next_page_number }})">&raquo;</button>
+                        </li>
+                    {% else %}
+                        <li class="disabled page-item"><span class="page-link">&raquo;</span></li>
+                    {% endif %}
+                </ul>
+            </nav>
+{% endif %}
+<table class="table table-striped">
```

Comment:
Just curious: would this still work / would it make sense if you separate this table into its own template and include it here? Same question for other tables. Again: I have no strong preference, just brainstorming.

Reply:
Yes, it would work but we would end up increasing the layers in the nested requests. This is not very recommended because it makes maintenance difficult. 

>Same question for other tables.

We at most places are using a separate table loaded through an Ajax-request (search. It is also almost the case here- just that we have filters and pagination also on the same page. I am not sure if we should go for a third layer.